### PR TITLE
Add interop with the worker_threads module

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

File: node_interop/test/promises_test.dart [hunk 1]
```diff
@@ -56,6 +56,14 @@ void main() {
...
   });
 
+  test('futureToPromise', () {
+    final JsPromises js = require(promises);
+    var future = Future.value('Yes');
+    var promise = futureToPromise(future);
+    var promise2 = js.receivePromise(promise);
+    expect(promiseToFuture(promise2), completion('YesYesYes'));
+  });
+
```
Comment (pulyaevskiy) [inline node_interop/test/promises_test.dart]: This looks like a duplicate of a test case from above?

**Review by pulyaevskiy — COMMENTED** [review]
LGTM, thanks for adding this! Just have a minor question in tests.

File: node_interop/test/promises_test.dart (hunk 1 above)
Comment (nex3) [reply node_interop/test/promises_test.dart]: Oops, that's a leftover from a change I ended up not making.

**Comment by pulyaevskiy** [comment]
Let me know if is this ready for merge or you were planning to push more updates. LGTM, otherwise.

**Comment by nex3** [comment]
Should be good to merge.

**Review by pulyaevskiy — APPROVED** [review]
LGTM, thanks!

**Comment by pulyaevskiy** [comment]
Released both changes as node_interop 1.2.0.

**Pull request merged**